hm-module.nix

```
@@ -76,9 +78,9 @@ in
       home.packages = [ cfg.package ];
 
       xdg.configFile = {
-        "zzz/config.yaml" = {
+        "zzz/config.yaml" = mkIf (!((cfg.settings == {}) && (cfg.settings == {}))) {
```

This should be an or, since if either are not empty we want to create the file.
```suggestion
        "zzz/config.yaml" = mkIf (!((cfg.settings == {}) || (cfg.settings == {}))) {
```

---

I forgot to mention that also means the `!` will have to move

---

don;t these two ways of writing the condition get the same result:
if the required result is that either should be nonempty inorder to create the config:
```nix
(cfg.settings != {}) || (cfg.colors != {})
# will return true if either is nonempty and return false if both are empty
! ( (cfg.settings == {}) && (cfg.settings == {}) )
# will return !true (false) if both are empty and !false (true) if one is nonempty
```
the real question is which way do you prefer to write it?

---

also i accidently wrote `cfg.settings` twice. so ill have to fix that

---

i actually prefer your way (the first).

---

Oh yeah. I guess you are right, feels a bit like cognitive overload though.